### PR TITLE
feat: add bridge_amino and test

### PR DESCRIPTION
**Pull request by mhagel**
Closes: #60 

- Adds AminoConverter for MsgBridge

### CODE REVIEW

File: packages/api/e2e/amino.spec.ts [hunk 1]
```diff
@@ -264,6 +265,24 @@ xdescribe('RegenApi with tendermint connection - Amino Tests', () => {
...
+          {
+            batchDenom: TEST_BRIDGE_BATCH_DENOM,
+            amount: MIN_CREDIT_AMOUNT,
+          },
+        ],
+      });
+
+      await runAminoTest(msgClient, TEST_MSG_BRIDGE);
+    });
   });
```
Comment (mhagel) [inline packages/api/e2e/amino.spec.ts:286]: I have tested this against redwood locally, including all negative tests I could think of, but we will need to implement #59 to run it in CI.

https://redwood.regen.aneka.io/txs/C70A43E236BAAB1B2DD2D352A03CBB72D8DF307EB7F03D61905315B26324C738

File: packages/api/src/tx/regen-message-type-registry.ts [hunk 2]
```diff
@@ -25,7 +25,7 @@ import {
 } from '../generated/regen/ecocredit/marketplace/v1/tx';
 import { GeneratedType } from '@cosmjs/proto-signing';
 
-const ecocreditRegistry: Array<[string, GeneratedType]> = [
+const ecocreditRegistry: ReadonlyArray<[string, GeneratedType]> = [
```
Comment (mhagel) [inline packages/api/src/tx/regen-message-type-registry.ts:28]: `ReadonlyArray` is more in line with the established convention for type registries.

File: packages/api/src/tx/modules/ecocredit/v1/bridge_amino.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
+import { AminoMsg } from '@cosmjs/amino';
+import { AminoConverter } from '@cosmjs/stargate';
+import { MsgBridge } from '../../../../generated/regen/ecocredit/v1/tx';
+import { Credits } from '../../../../generated/regen/ecocredit/v1/types';
+import { AminoCredits } from './msg_cancel';
+
+const msgBridgeAminoType = 'regen.core/MsgBridge';
```
Comment (blushi) [inline packages/api/src/tx/modules/ecocredit/v1/bridge_amino.ts:7]: just a heads up that we'll need to remove "core" as soon as we move to v5

File: packages/api/src/tx/msg.ts [hunk 4]
```diff
@@ -37,7 +37,7 @@ export async function setupTxExtension(
   messageTypeRegistry.forEach((value, key) => {
     customRegistry.push([`/${key}`, value]);
   });
-  regenRegistry.forEach(([key, value]) => {
+  regenTypeRegistry.forEach(([key, value]) => {
     customRegistry.push([`/${key}`, value]);
   });
```
Comment (blushi) [inline packages/api/src/tx/msg.ts:42]: not related to this PR, but why do we need this?
`messageTypeRegistry.forEach` above is already registering all the types

File: packages/api/src/tx/msg.ts [hunk 5]
```diff
@@ -37,9 +36,6 @@ export async function setupTxExtension(
   messageTypeRegistry.forEach((value, key) => {
     customRegistry.push([`/${key}`, value]);
   });
-  regenRegistry.forEach(([key, value]) => {
```
Comment (mhagel) [inline packages/api/src/tx/msg.ts:40]: deleting this. As @blushi pointed out, the `messageTypeRegistry` already registers all the custom regen msgs.

File: packages/api/src/tx/msg.ts (hunk 4 above)
Comment (mhagel) [reply packages/api/src/tx/msg.ts]: deleted

File: packages/api/src/tx/modules/ecocredit/v1/bridge_amino.ts (hunk 3 above)
Comment (mhagel) [reply packages/api/src/tx/modules/ecocredit/v1/bridge_amino.ts:7]: Yes we can do it in https://github.com/regen-network/regen-js/issues/62 just to keep it consistent

**Review by blushi — APPROVED** [review]

**Review by flagrede — APPROVED** [review]
Looks good!

**Pull request merged**